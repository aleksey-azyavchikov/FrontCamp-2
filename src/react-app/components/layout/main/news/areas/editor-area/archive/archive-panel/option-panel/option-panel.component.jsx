

import React, { PureComponent } from "react";

import "./option-panel.component.scss";

export class OptionPanel extends PureComponent {

    getIconClasses(icon, currentMode, mode) {
        const defaultClasses = `panel-option oi ${icon} icon`;
        const selectedClass = currentMode === mode ? "selected-mode" : "";
        return `${defaultClasses} ${selectedClass}`

    }

    render() {
        const { currentMode, componentMode, iconClass, onModeClick } = this.props;

        return (
            <li class="list-inline-item">
                <span
                    id={componentMode}
                    onClick={() => onModeClick(componentMode)}
                    class={this.getIconClasses(iconClass, currentMode, componentMode)}>
                </span>
            </li>
        );
    }

}