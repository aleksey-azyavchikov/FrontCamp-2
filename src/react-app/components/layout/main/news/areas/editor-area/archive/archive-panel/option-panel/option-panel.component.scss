.icon:hover, .selected-mode {
    color: rgb(134, 217, 230);
    cursor: pointer;
}
