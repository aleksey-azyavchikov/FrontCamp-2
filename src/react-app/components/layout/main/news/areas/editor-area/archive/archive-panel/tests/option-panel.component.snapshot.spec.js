import React from "react";
import { OptionPanel } from "../option-panel/option-panel.component";
import renderer from "react-test-renderer";

describe("Compoent: OptionPanel", () => {
    let props;
    beforeAll(() => {
    })

    beforeEach(() => {
        props = { }
    })

    it("Display: initial", () => {
        const rendered = renderer.create(
            <OptionPanel {...props}/>
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    })
}) 
