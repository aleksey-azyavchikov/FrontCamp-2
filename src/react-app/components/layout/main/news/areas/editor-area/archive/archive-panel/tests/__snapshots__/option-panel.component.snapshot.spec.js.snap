// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compoent: OptionPanel Display: initial 1`] = `
<li
  className="list-inline-item"
>
  <span
    className="panel-option oi undefined icon selected-mode"
    id={undefined}
    onClick={[Function]}
  />
</li>
`;
